Objects/funcobject.c

```diff
@@ -942,17 +942,12 @@ PyTypeObject PyFunction_Type = {
 static int
 functools_copy_attr(PyObject *wrapper, PyObject *wrapped, PyObject *name)
 {
-    PyObject *value = PyObject_GetAttr(wrapped, name);
-    if (value == NULL) {
-        if (PyErr_ExceptionMatches(PyExc_AttributeError)) {
-            PyErr_Clear();
-            return 0;
-        }
-        return -1;
+    PyObject *value;
+    int res = _PyObject_LookupAttr(wrapped, name, &value);
+    if (res > 0) {
```

Comment:
Perhaps it will be better:
```suggestion
    if (value != NULL) {
```